### ISSUE TITLE
nixos.tests.printing.*-linux (cups) is failing

### ISSUE THREAD

**Issue by obadz** [NixOS/nixpkgs#14748]
Probably linked to `closure-size`.

http://hydra.nixos.org/build/34566747/nixlog/20/tail-reload
```
error: unit ‘cups.service’ is inactive and there are no pending jobs
unit ‘cups.service’ is inactive and there are no pending jobs
```

@vcunat @edolstra @domenkozar @joachifm @abbradar 

Holding up #14595

**Comment by joachifm** [comment]
I'm quite sure cups isn't even scheduled for execution. If you read the test output, there's nary a mention of cups.

If you simply set `services.printing.enable = true`, you end up with a `cups.service` that lacks an `ExecStart` directive, making it an invalid service unit. systemd, being helpful as always, ignores invalid units and carries on as if nothing happened ...

**Comment by joachifm** [comment]
4ac95ab3282a5aa364c8c1e51b9637b38729a3a7 removed explicit `ExecStart` from the unit, relying instead on the upstream service units, which I guess are no longer available due to output splitting.

**Comment by joachifm** [comment]
Okay, I think I figured it out. The problem is that `cupsd -l` forks into the background, whereas the upstream service expects it to remain in the foreground, consequently, systemd will think that the service has completed and thus record it as inactive.

**Comment by joachifm** [comment]
To be clear, `cupsd -l` is NOT supposed to fork into the background, but does so because it hasn't been compiled with systemd support for some reason.

**Comment by dezgeg** [comment]
Probably caused by the systemd split then. (AFAIU it was very painful to get that split correctly).

**Comment by joachifm** [comment]
I've fixed it, will push shortly

**Comment by joachifm** [comment]
Now, of course the test is STILL broken, but at least it doesn't fail due to `waitForUnit`

**Comment by joachifm** [comment]
... and at least the client service should be functional (as in running & responding to commands, which it didn't before these patches).

**Comment by dezgeg** [comment]
It could be some of these invalid references to `cups.dev`: https://github.com/dezgeg/nixpkgs/commit/8c5936ae8788f9e2d1b97fe984a66afb0e33beb9

**Comment by joachifm** [comment]
The test now fails because the client node gets a 404 from the server node.

**Comment by aszlig** [comment]
@joachifm: Not only on your machine but also on Hydra: https://hydra.nixos.org/build/34632091/nixlog/31/raw

**Comment by joachifm** [comment]
Well, that's progress ...

**Comment by joachifm** [comment]
Great ... removing `--fail` from the curl calls magically makes it work. But then there are still more errors.

**Comment by vcunat** [comment]
Maybe related: cd8b383f.

**Comment by vcunat** [comment]
Actually, this is more direct: https://github.com/NixOS/nixpkgs/issues/14782.

**Comment by obadz** [comment]
@vcunat, looks like that did not fix it :-(

**Comment by vcunat** [comment]
Yes, the test still fails, but I merged anyway to fix the regression of doc location #14782.

Has anyone managed to successfully run the test on some recent branch? (or managed to find what's the matter with it?)
